component-test/test_dbus_proxy.py

```diff
@@ -155,6 +155,152 @@
     "opath_1": stubs.OPATH_1
 })
 
+CONF_WHITELIST_DIRECTIONS = """
+{{
+    "dbus-gateway-config-session": [{{
+        "direction": "*",
+        "interface": "*",
+        "object-path": "*",
+        "method": "*"
+    }},
+    {{
+        "direction": "outgoing",
+        "interface": "{iface}.{extension_1}",
+        "object-path": "*",
+        "method": "*"
+    }}],
+    "dbus-gateway-config-system": []
+}}
+""".format(**{
+    "iface": stubs.IFACE_1,
+    "extension_1": stubs.EXT_1
+})
+
+CONF_WHITELIST_DIRECTIONS_REVERSE = """
+{{
+    "dbus-gateway-config-session": [{{
+        "direction": "outgoing",
+        "interface": "{iface}.{extension_1}",
+        "object-path": "*",
+        "method": "*"
+    }},
+    {{
+        "direction": "*",
+        "interface": "*",
+        "object-path": "*",
+        "method": "*"
+    }}],
+    "dbus-gateway-config-system": []
+}}
+""".format(**{
+    "iface": stubs.IFACE_1,
+    "extension_1": stubs.EXT_1
+})
+
+CONF_WHITELIST_PATH = """
+{{
+    "dbus-gateway-config-session": [{{
+        "direction": "*",
+        "interface": "*",
+        "object-path": "*",
+        "method": "*"
+    }},
+    {{
+        "direction": "incoming",
+        "interface": "{iface}.{extension_1}",
+        "object-path": "/a/path/to/unavailable/directory/",
+        "method": "*"
+    }}],
+    "dbus-gateway-config-system": []
+}}
+""".format(**{
+    "iface": stubs.IFACE_1,
+    "extension_1": stubs.EXT_1
+})
+
+CONF_WHITELIST_PATH_REVERSE = """
+{{
+    "dbus-gateway-config-session": [{{
+        "direction": "incoming",
+        "interface": "{iface}.{extension_1}",
+        "object-path": "/a/path/to/unavailable/directory/",
+        "method": "*"
+    }},
+    {{
+        "direction": "*",
+        "interface": "*",
+        "object-path": "*",
+        "method": "*"
+    }}],
+    "dbus-gateway-config-system": []
+}}
+""".format(**{
+    "iface": stubs.IFACE_1,
+    "extension_1": stubs.EXT_1
+})
+
+
+class TestWhitelisting(object):
```

Comment:
I think the other previously existing whitelisting tests should be moved to this suite now. It might only be the "test contradicting rules" that is relevant. But take a look through the existing tests and structure to make sure it all looks coherent please.